neo4j/internal/connector/connector.go

```diff
@@ -137,17 +139,35 @@ func (c Connector) createConnection(ctx context.Context, address string) (net.Co
 }
 
 func (c Connector) tlsConfig(serverName string) *tls.Config {
-	var config *tls.Config
-	if c.Config.TlsConfig == nil {
-		//lint:ignore SA1019 RootCAs is supported until 6.0
-		config = &tls.Config{RootCAs: c.Config.RootCAs}
-	} else {
-		config = c.Config.TlsConfig
+	// Start with the provided TlsConfig or initialize a new one if not provided.
+	config := c.Config.TlsConfig
+	if config == nil {
+		config = &tls.Config{
+			// Use RootCAs from the connector's config.
+			//lint:ignore SA1019 RootCAs is supported until 6.0
+			RootCAs: c.Config.RootCAs,
+			// It's safe to set MinVersion and other settings here since we're initializing a new config.
+			MinVersion: tls.VersionTLS12,
+		}
 	}
-	if config.MinVersion == 0 {
+
+	// Ensure MinVersion is set to at least TLS 1.2.
```

Comment:
Should this be at least another PR? This is a change worth it's own point in the changelog. And might be perceived as too intrusive. Before, the user was allowed to pass in a custom `tls.Config` which the driver uses (almost) unchanged. With this change, we force it to use at least TLS1.2.

Reply:
Yeah, I think so, I'll revert it to the `config.MinVersion == 0` check and have a chat with you for future change 👍

Reply:
Docs state: 

```
	// MinVersion contains the minimum TLS version that is acceptable.
	//
	// By default, TLS 1.2 is currently used as the minimum when acting as a
	// client, and TLS 1.0 when acting as a server. TLS 1.0 is the minimum
	// supported by this package, both as a client and as a server.
	//
	// The client-side default can temporarily be reverted to TLS 1.0 by
	// including the value "x509sha1=1" in the GODEBUG environment variable.
	// Note that this option will be removed in Go 1.19 (but it will still be
	// possible to set this field to VersionTLS10 explicitly).
```

So this might be a pointless bit of logic anyways?

Reply:
There is a slight difference. If we remove it all together, the `Config.TlsConfig` docs should specify that we leave it up to go's tls package to choose a default min. TLS version (might change in the future :shrug:.

Reply:
Cool I've reverted it to the original logic for now and will tackle this later 👍